content/en/blog/travelogue/index.md

```diff
@@ -0,0 +1,421 @@
+
+I had had quite a long time the idea of using gRPC for the Cloud Agent
+controller.  My core idea was to get rid of the EA because, currently, it was
+just an onboarding tool. The wallet had, included only the pairwise DID to its
+cloud agent, nothing else. The actual wallet (we called it worker edge agent
+wallet) was the real wallet, where the VCs were. I went thru
+many similar protocols until I found FIDO UAF. The protocol is similar to
+DIDComm's pairwise protocol, but it's not symmetric. Another end is the server,
+and the other has the authenticator -- the cryptographical root of
+trust.
+
+When I presented an internal demo of the gRPC with the JWT authorization and
+explained that authentications would be FIDO2 WebAuth, we were ready to start
+the architecture transition. Everything was still good when I implemented the
+first FIDO server with the help of Duo Labs Go packages. Our FIDO2 server was
+now capable to allocated cloud agents. But there was one missing part I was
+hoping someone in the OSS community would implement until we needed it. It was a
+headless WebAuthn/UAF authenticator for those issuers/verifiers running as
+service agents. How to onboard them, and how they would access the agency's
+resources with the same JWT authorization? To allow us to proceed, we added
+support to get JWT by our old API. It's was only intermediated solution but
+served its purpose.
+
+##### Learnings when implementing the new architecture 
+
+- implicit JWT authorization helps gRPC usage a lot and simplifies it too.
+- gRPC streams and Go's channel is just excellent together.
+- You should use pre-generated wallet keys for indy wallets.
+- We can integrate performance and scalability tests into CI.
+- gRPC integration and unit testing could be done in the same way as with HTTP
+stack in Go, i.e. inside a single process that can play both client and server.
+
+##### Highlights of the end of the year 2020
+
+We started to build for new SA architecture and allowed both our APIs to
+existing.  WebAuth server, headless authenticator, and Vault first versions were
+now ready.  Also, I did the first version of a state machine for service agent
+implementation. We had an option to use immuDB instead of Plenum ledger.
+
+### 2021/H1
+
+{{< imgproc spring21.png Resize "2000x">}}
+<em>2021/H1</em>
+{{< /imgproc >}}
+
+Now we have an architecture that we can live with. All the important elements
+are in place. Now we just clean it up.
+
+##### Summary of Spring 2021 Results
+
+Until the summer, the most important results have been:
+- Headless WebAuthn authenticator 
+- React-based Web Wallet
+- Lots of documentation and examples
+- Agency's gRPC API v1
+- Polyglot implementations gRPC: TypeScript, Go, JavaScript
+- New toolbox both Web and Command-line 
+- The full OSS release 
+
+As said, all of the important elements are in place. However, our solution is
+based on `libindy`, which is interesting where the original contributor
```

Comment:
I would reformat this somehow. _Our solution is currently based on libindy, which makes the future of the library interesting. Many from the Aries community are moving to shared libraries whereas the original contributor has chosen its own path._

Reply:
nice, and totally agree. actually moving a little bit back to the original, maybe I trusted too much Grammarly on this :-)